src/fparser/two/tests/test_parser.py

```diff
@@ -84,3 +84,47 @@ class do not affect current calls.
     with pytest.raises(ValueError) as excinfo:
         parser = ParserFactory().create(std="invalid")
         assert "is an invalid standard" in str(excinfo.value)
+
+
+def test_deepcopy():
+    """
+    Test that we can deepcopy a parsed fparser tree.
+    """
+
+    f90_source = """
+program main
+
+  implicit none
+
+end
+"""
+    parser = ParserFactory().create(std="f2008")
+    reader = FortranStringReader(f90_source)
+    ast = parser(reader)
+
+    import copy
+
+    _ = copy.deepcopy(ast)
```

Comment:
Please add some minimal check here that the copy has the same form as the original. Same root node, same number of nodes perhaps? You could use the `walk()` utility to help.

Reply:
See above. It's done.

Reply:
I added more code to this example.
I also changed the string comparison to do a string comparison for the following reasons:

It turns out that this assertion is always valid:
t1 = "a"
t2 = "a"
assert t1 is t2
assert id(t1) == id(t2)